src/ctapipe/io/tests/test_simteleventsource.py

```diff
@@ -236,6 +236,24 @@ def test_calibration_events():
             assert event.index.event_id == expected_id
 
 
+def test_skip_r1_calibration():
+    n_expected = 5
+    with SimTelEventSource(
+        input_url=calib_events_path,
+        max_events=n_expected,
+        skip_calibration_events=False,
+        skip_r1_calibration=True,
+        focal_length_choice="EQUIVALENT",
+    ) as reader:
+        n_processed = 0
+        for event in reader:
+            n_processed += 1
+            assert np.issubdtype(event.r0.tel[1].waveform.dtype, np.uint16)
+            assert np.issubdtype(event.r1.tel[1].waveform.dtype, np.float32)
+            assert np.allclose(event.r0.tel[1].waveform, event.r1.tel[1].waveform)
```

Comment:
float32 can represent all values of uint16 exactly, so this should check actual equality not just `isclose`.

Also, for better failure messages, use the functions in `numpy.testing` instead of `assert ...`